pkg/steve/pkg/devicetemplaterevisionapi/store.go

```diff
@@ -0,0 +1,259 @@
+package devicetemplaterevisionapi
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cnrancher/edge-api-server/pkg/apis/edgeapi.cattle.io/v1alpha1"
+	"github.com/cnrancher/edge-api-server/pkg/auth"
+	controller "github.com/cnrancher/edge-api-server/pkg/generated/controllers/edgeapi.cattle.io/v1alpha1"
+	"github.com/cnrancher/edge-api-server/pkg/util"
+	"github.com/rancher/steve/pkg/accesscontrol"
+	"github.com/rancher/steve/pkg/schemaserver/types"
+	"github.com/rancher/wrangler/pkg/data/convert"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+)
+
+type Store struct {
+	types.Store
+	asl                              accesscontrol.AccessSetLookup
+	client                           dynamic.Interface
+	ctx                              context.Context
+	auth                             auth.Authenticator
+	deviceTemplateController         controller.DeviceTemplateController
+	deviceTemplateRevisionController controller.DeviceTemplateRevisionController
+}
+
+const (
+	templateRevisionDeviceTypeName = "edgeapi.cattle.io/device-template-revision-type"
+	templateRevisionOwnerName      = "edgeapi.cattle.io/device-template-revision-owner"
+	templateNameLabel              = "edgeapi.cattle.io/device-template-name"
+)
+
+func (s *Store) Create(apiOp *types.APIRequest, schema *types.APISchema, data types.APIObject) (types.APIObject, error) {
+	var deviceTemplateRevision v1alpha1.DeviceTemplateRevision
+	err := convert.ToObj(data.Data(), &deviceTemplateRevision)
+	if err != nil {
+		logrus.Errorf("failed to convert device template revision data, error: %s", err.Error())
+		return data, err
+	}
+
+	if err := ValidateTemplateRequest(&deviceTemplateRevision.Spec); err != nil {
+		logrus.Errorf("invalid device template revision request, error: %s", err.Error())
+		return data, err
+	}
+
+	if err := ValidTemplateSpec(s.ctx, &deviceTemplateRevision, s.client); err != nil {
+		logrus.Errorf("valid template spec error: %s", err.Error())
+		return data, err
+	}
+
+	deviceTemplate, err := ValidateDeviceTemplateIsExist(s.ctx, &deviceTemplateRevision, s.deviceTemplateController)
```

Comment:
this may not necessarily be required since the API creation of device template and device revision can happen simultaneously.

Reply:
> this may not necessarily be required since the API creation of device template and device revision can happen simultaneously.

I think it was necessary for him to check when he updated the deviceTemplateName field of the revision